AWSAuthSDK/Sources/AWSMobileClient/AWSMobileClient.swift

```diff
@@ -459,13 +462,24 @@ final public class AWSMobileClient: _AWSMobileClient {
                     signInInfo[self.TokenKey] = session.accessToken!.tokenString
                     signInInfo[self.ProviderKey] = "OAuth"
                     
-                    self.performHostedUISuccessfulSignInTasks(disableFederation: hostedUIOptions.disableFederation, session: session, federationToken: federationToken!, federationProviderIdentifier: federationProviderIdentifier, signInInfo: &signInInfo)
-                    completionHandler(.signedIn, nil)
-                    if self.pendingGetTokensCompletion != nil {
-                        self.tokenFetchLock.leave()
-                    }
-                    self.pendingGetTokensCompletion?(self.getTokensForCognitoAuthSession(session: session), nil)
-                    self.pendingGetTokensCompletion = nil
+                    self.performHostedUISuccessfulSignInTasks(disableFederation: hostedUIOptions.disableFederation,
+                                                              session: session,
+                                                              federationToken: federationToken!,
+                                                              federationProviderIdentifier: federationProviderIdentifier,
+                                                              signInInfo: &signInInfo).continueWith(block: { task -> Any? in
```

Comment:
nit: use trailling closure syntax for `continueWith`

Reply:
Added